Library/Homebrew/utils/gems.rb

```diff
@@ -51,8 +51,13 @@ def odie_if_defined(message)
   end
 
   def setup_gem_environment!(gem_home: nil, gem_bindir: nil, setup_path: true)
+    library_path = if ENV["HOMEBREW_STACKPROF"]
```

Comment:
I think a comment explaining why we need to do this would be good:

```suggestion
    # When `HOMEBREW_STACKPROF` is set, this method is called before `HOMEBREW_LIBRARY_PATH` is defined
    library_path = if ENV["HOMEBREW_STACKPROF"]
```